packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -0,0 +1,274 @@
+   *
+   * @default - None
+   */
+  readonly tags?: { [name: string]: string };
+}
+
+/**
+ * NodeGroup properties interface
+ */
+export interface NodegroupProps extends NodegroupOptions {
+  /**
+   * Cluster resource
+   * [disable-awslint:ref-via-interface]"
+   */
+  readonly cluster: Cluster;
+}
+
+/**
+ * The Nodegroup resource class
+ */
+export class Nodegroup extends Resource implements INodegroup {
+  /**
+   * Import the Nodegroup from attributes
+   */
+  public static fromNodegroupName(scope: Construct, id: string, nodegroupName: string): INodegroup {
+    class Import extends Resource implements INodegroup {
+      public readonly nodegroupName = nodegroupName;
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * ARN of the nodegroup
+   *
+   * @attribute
+   */
+  public readonly nodegroupArn: string;
+  /**
+   * Nodegroup name
+   *
+   * @attribute
+   */
+  public readonly nodegroupName: string;
+  /**
+   * the Amazon EKS cluster resource
+   *
+   * @attribute ClusterName
+   */
+  public readonly cluster: Cluster;
+  /**
+   * IAM role of the instance profile for the nodegroup
+   */
+  public readonly role: IRole;
+
+  private readonly desiredSize: number;
+  private readonly maxSize: number;
+  private readonly minSize: number;
+
+  constructor(scope: Construct, id: string, props: NodegroupProps ) {
+    super(scope, id, {
+      physicalName: props.nodegroupName
+    });
+
+    this.cluster = props.cluster;
+
+    this.desiredSize = props.desiredSize ?? props.minSize ?? 2;
+    this.maxSize = props.maxSize ?? this.desiredSize;
+    this.minSize = props.minSize ?? 1;
+
+    if (this.desiredSize > this.maxSize) {
+      throw new Error(`Desired capacity ${this.desiredSize} can't be greater than max size ${this.maxSize}`);
+    }
+    if (this.desiredSize < this.minSize) {
+      throw new Error(`Minimum capacity ${this.minSize} can't be greater than desired size ${this.desiredSize}`);
+    }
+
+    if (!props.nodeRole) {
+      const ngRole = new Role(this, 'NodeGroupRole', {
+        assumedBy: new ServicePrincipal('ec2.amazonaws.com')
+      });
+
+      ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'));
+      ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'));
+      ngRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'));
+      this.role = ngRole;
+    } else {
+      this.role = props.nodeRole;
+    }
+
+    const resource = new CfnNodegroup(this, 'Resource', {
+      clusterName: this.cluster.clusterName,
+      nodegroupName: props.nodegroupName,
+      nodeRole: this.role.roleArn,
+      subnets: this.cluster.vpc.selectSubnets(props.subnets).subnetIds,
+      amiType: props.amiType,
+      diskSize: props.diskSize,
+      forceUpdateEnabled: props.forceUpdateEnabled ?? true,
+      instanceTypes: props.instanceType ? [props.instanceType.toString()] : undefined,
```

Comment:
Do they really support multiple instance types? Do we want to support that too?

Reply:
They support one instance type at this moment, but the property type in the cfn spec is list. But it makes sense to support list as well in CDK. Let me revise it.

Reply:
However, if we support multiple instance types then we have to restrict user only provide one instance type in the list, otherwise throw an error. What do you think?

Reply:
Maybe we can still make this property `instanceType[]` and note this the doc string that we only select the first one at this moment. If undefined, the default value would be `t3.medium` which is the [default value in cfn](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-nodegroup.html#cfn-eks-nodegroup-instancetypes).

Now we have another question, previously the `new eks.Cluster()` with no specific instance type specified will make `m5.large` as default, now would be `t3.medium`. Should we make it `m5.large` for consistency?

Reply:
No no let’s just support a single instance type. When this changes, we can introduce “instanceTypes” and an additional way to specify this.